lib/SPIRV/SPIRVTypeScavenger.cpp

```diff
@@ -226,6 +231,20 @@ void SPIRVTypeScavenger::deduceFunctionType(Function &F) {
       DeducedTypes[Arg] = Ty;
   }
 
+  // The first non-sret argument of block_invoke functions is the block capture
+  // struct, which should be passed as an i8*.
+  static Regex BlockInvokeRegex("^(__.+)?_block_invoke(_[0-9]+)?(_kernel)?$");
```

Comment:
To clarify, why do we need it be static?

Reply:
This doesn't need to be static, but it helps if it is static because re-building the regex each time the function is called can be wasteful.

It looks like I missed marking this const though; I've added a small update for that.